test/test_common/utility.cc

```diff
@@ -117,12 +117,16 @@ std::vector<std::string> TestUtility::listFiles(const std::string& path, bool re
   dirent* entry;
   while ((entry = readdir(dir)) != nullptr) {
     std::string file_name = fmt::format("{}/{}", path, std::string(entry->d_name));
-    if (recursive && entry->d_type == DT_DIR && std::string(entry->d_name) != "." &&
+    struct stat stat_result;
+    int rc = ::stat(file_name.c_str(), &stat_result);
+    EXPECT_EQ(rc, 0);
```

Comment:
Nit: I would make this `ASSERT_EQ`, since if this fails, the code below is using junk.